hugo/content/docs/langium-overview.md

```diff
@@ -0,0 +1,103 @@
+
+Langium accomplishes this feat by using the concept of 'scoping'. You likely know scopes from programming, where some variables are only available from certain scopes:
+
+```ts
+let x = 42;
+x = 3; // References the `x` defined in the previous line
+```
+
+```ts
+if (something) {
+    let y = 42;
+}
+y = 3; // Cannot link, `y` isn't in any of the available scopes
+```
+
+The same occurs in Langium. To enable more complex scoping behavior, you can create a custom domain scoping. For example, common object-oriented languages need a more involved scoping mechanism to resolve references to fields and methods of a class:
+
+```ts
+class X {
+    y(): void { ... }
+}
+
+const instance = new X(); // Symbol `X` is in the local scope
+instance.y(); // Symbol `y` exists in the scope of the `X` class
+```
+
+Once your domain specific scoping rules have been defined, Langium will take care of linking by itself, reporting any errors appearing 
+
+## Workspace Management
+
+Like with common modularized programming languages, domain logic written in your DSL will usually be split across multiple files to facilitate ease of use and maintenance efforts. This is also possible using Langium, which automatically tries to pick up any files belonging to your current project.
+
+When running a Langium based language in a [language server](https://microsoft.github.io/language-server-protocol/), all files in your workspace, i.e. the folder containing your current project, matching your DSL Id will automatically be picked up and processed. In addition, any changes in your workspace will be handled as well. Dealing with added, changed or deleted files in a workspace with multiple hundreds of files can become complicated and decrease performance drastically if not done correctly. Langium employs heuristics to only invalidate and recompute what is actually necessary.
+
+The workspace management also keeps track of the global scope. This allows users of your DSL to reference elements across files within the same workspace.
+
+## Editing Support
+
+The Langium framework is deeply integrated with the [language server protocol](https://microsoft.github.io/language-server-protocol/) (LSP). The LSP aims to reduce the integration efforts when designing a language by providing an interface that all IDEs can use to provide editing support. This allows Langium based languages to easily interact with common IDEs and editors with LSP support, including Visual Studio Code, Eclipse, IntelliJ and many more.
+
+The LSP includes commonly used language features, such as code completion, custom validations/diagnostics, finding references, formatting and many more, allowing for deep IDE integration without binding your language to a single IDE. Langium offers out-of-the-box support for most of these language features, with additional extension points for your domain specific requirements.
```

Comment:
suggestion, break up into two sentences.

> formatting and many more, allowing...

formatting and many more **. This allows**...